packages/patternfly-4/react-core/src/components/Nav/NavGroup.js

```diff
@@ -0,0 +1,90 @@
+import React from 'react';
+import styles from '@patternfly/patternfly-next/components/Nav/styles.css';
+import a11yStyles from '@patternfly/patternfly-next/utilities/Accessibility/styles.css';
+import { css } from '@patternfly/react-styles';
+import PropTypes from 'prop-types';
+import NavToggle from './NavToggle';
+import { AngleRightIcon } from '@patternfly/react-icons';
+
+const propTypes = {
+  id: PropTypes.string.isRequired,
+  title: PropTypes.string.isRequired,
+  ariaLabel: PropTypes.string.isRequired,
+  expandable: PropTypes.bool.isRequired,
+  children: PropTypes.node.isRequired,
+  innerClassName: PropTypes.string.isRequired,
+  groupId: PropTypes.number.isRequired,
+  activeGroup: PropTypes.number.isRequired,
+  activeItem: PropTypes.number.isRequired
+};
+
+const NavGroup = ({
+  id,
+  title,
+  ariaLabel,
+  expandable,
+  children,
+  innerClassName,
+  groupId,
+  activeGroup,
+  activeItem,
+  ...props
+}) => {
+  if (expandable) {
+    return (
+      <NavToggle defaultValue={groupId === activeGroup}>
+        {({ value: isExpanded, toggle }) => (
+          <li
+            className={css(
+              styles.navItem,
+              isExpanded && styles.modifiers.expanded,
+              activeGroup === groupId && styles.modifiers.current,
+              innerClassName
+            )}
+            onClick={toggle}
+            {...props}
+          >
+            <a
+              className={css(styles.navLink)}
+              id={ariaLabel ? '' : id}
```

Comment:
I don't understand why we would want to assign the ariaLabel to the id here?

Reply:
Depending on if aria-label is passed in, it will be used as text in an invisible <h2> section below and this id will not be added to the dom. If a custom aria-label is not passed in, then the id is set here and referenced for aria-labelledby